python/ray/serve/pipeline/conftest.py

```diff
@@ -12,3 +13,39 @@ def shared_ray_instance():
     else:
         # Overriding task_retry_delay_ms to relaunch actors more quickly.
         yield ray.init(num_cpus=36, _system_config={"task_retry_delay_ms": 50})
+
+
+@pytest.fixture
+def ray_shutdown():
+    yield
+    serve.shutdown()
+    ray.shutdown()
+
+
+@pytest.fixture(scope="session")
+def _shared_serve_instance():
+    # Note(simon):
+    # This line should be not turned on on master because it leads to very
+    # spammy and not useful log in case of a failure in CI.
+    # To run locally, please use this instead.
+    # SERVE_LOG_DEBUG=1 pytest -v -s test_api.py
+    # os.environ["SERVE_LOG_DEBUG"] = "1" <- Do not uncomment this.
+
+    # Overriding task_retry_delay_ms to relaunch actors more quickly
+    ray.init(
+        num_cpus=36,
+        namespace="default_test_namespace",
+        _metrics_export_port=9999,
+        _system_config={"metrics_report_interval_ms": 1000, "task_retry_delay_ms": 50},
+    )
+    yield serve.start(detached=True)
+
+
+@pytest.fixture
+def serve_instance(_shared_serve_instance):
+    yield _shared_serve_instance
+    # Clear all state between tests to avoid naming collisions.
+    for deployment in serve.list_deployments().values():
+        deployment.delete()
+    # Clear the ServeHandle cache between tests to avoid them piling up.
```

Comment:
can we import these from the other file?

Reply:
don't want 2 copies of them lying around and getting out of sync

Reply:
I can give that a try. Afiak it's kind of anti pattern of we import functions from `conftest`, but should be ok if `conftest` import from others.